### PR TITLE
feat: display ordinals

### LINKED ISSUE #3090: Display Ordinal inscriptions alongside Stacks NFTs
To complement the release of https://github.com/hirosystems/stacks-wallet-web/issues/1028

### PR DESCRIPTION
**Pull request by edu-stx**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4206328875).

Closes https://github.com/hirosystems/stacks-wallet-web/issues/3090

Left in some commented out code while I figure out how to create testnet ordinals. Will remove once sure the feature is working.

Looks something like,

image

### CODE REVIEW

**Review by kyranjamie — COMMENTED** [review]
Good job Edu. 

I recommend making ordinals a feature in it's own right in `features/ordinals`.

I know we're trying to move fast, but would be great to also make layout component and put the query it it's own hook.

**Comment by markmhendrickson** [comment]
Looks great! These particular inscriptions are going to be collector items in the not-too-far future 😂

How's it format in the thinner extension dropdown viewport? Two per row or something?

**Comment by markmhx** [comment]
Looks great! These particular inscriptions are going to be collector items in the not-too-far future 😂

How's it format in the thinner extension dropdown viewport? Two per row or something?

**Comment by edu-stx** [comment]
@markmhx yes :) Updated original post.

File: src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { useQuery } from '@tanstack/react-query';
```
Comment (kyranjamie) [inline src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts]: filename against convention

File: src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { ordApiXyzGetInscriptionByAddressSchema } from '../types';
+
+export function useGetOrdinalsQuery() {
+  // TODO: although this hook returns a btc address, it's not the correct one for ordinals.
+  const bitcoinAddress = useCurrentBtcAccountAddressIndexZero();
+
+  return useQuery(['ordinals', bitcoinAddress] as const, async ({ queryKey }) => {
+    const [_, address] = queryKey;
+
+    if (!address) return [];
```
Comment (kyranjamie) [inline src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts]: can use the enabled prop to turn off the query when there's no address

File: src/app/features/collectibles/ordinals.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { Box, Grid, Stack, Text, color } from '@stacks/ui';
+
+// import { useQuery } from '@tanstack/react-query';
+import { openInNewTab } from '@app/common/utils/open-in-new-tab';
+import { Caption } from '@app/components/typography';
+
+import { useGetOrdinalsQuery } from './hooks/useGetOrdinalsQuery';
+
+interface Props {
```
Comment (kyranjamie) [inline src/app/features/collectibles/ordinals.tsx]:
```suggestion
interface OrdinalsProps {
```

File: src/app/features/collectibles/collectibles.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Box } from '@stacks/ui';
+
+import { Caption } from '@app/components/typography';
+
+import { useGetOrdinalsQuery } from './hooks/useGetOrdinalsQuery';
+import { Ordinals } from './ordinals';
+
+export function Collectibles() {
```
Comment (kyranjamie) [inline src/app/features/collectibles/collectibles.tsx:9]: Like this naming/feature abstraction 👍🏼

File: src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts (hunk 1 above)
Comment (edu-stx) [reply src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts]: slipped by, thanks :)

File: src/app/features/collectibles/hooks/use-get-ordinals-query.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useCurrentBtcAccountAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/bitcoin-account.hooks';
+
+import { ordApiXyzGetInscriptionByAddressSchema } from '../types';
+
+export function useGetOrdinalsQuery() {
```
Comment (fbwoolf) [inline src/app/features/collectibles/hooks/use-get-ordinals-query.ts]: Should this just live with our other queries in `query` folder?

Comment (edu-stx) [reply src/app/features/collectibles/hooks/use-get-ordinals-query.ts]: It can, what's the convention or criteria for where queries are placed?

Comment (kyranjamie) [reply src/app/features/collectibles/hooks/use-get-ordinals-query.ts]: Currently, we put them all under `src/app/queries`

File: src/app/features/collectibles/types.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import * as yup from 'yup';
```
Comment (kyranjamie) [inline src/app/features/collectibles/types.ts]: Filename is kind of confusing. No types declared here, only validation. `ordinal-validator` or similar?

File: src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/features/collectibles/hooks/useGetOrdinalsQuery.ts]: fix hasn't been committed yet*

File: src/app/features/collectibles/ordinals.tsx (hunk 3 above)
Comment (kyranjamie) [reply src/app/features/collectibles/ordinals.tsx]: I saw this was fixed in last push, but now reverted back to nameless prop?

File: src/app/features/collectibles/hooks/use-get-ordinals-query.ts (hunk 5 above)
Comment (edu-stx) [reply src/app/features/collectibles/hooks/use-get-ordinals-query.ts]: Ok, moved :) I saw `utils` files sprinkled throughout the queries folder, so created one for the ordinal validator, let me know if this works?

File: src/app/features/collectibles/ordinals.tsx (hunk 3 above)
Comment (edu-stx) [reply src/app/features/collectibles/ordinals.tsx]: I pushed the wrong commit, back to `OrdinalProps` now :)

**Comment by mica000** [comment]
@edu-stx V. excited!! So far I don't have any Ordinals so I only see the ghost button. What is the best way to see the display of the other images?

Smt to be improved: we are missing the hover and clicked states on the button
Screenshot 2023-02-17 at 21 29 06

https://www.figma.com/file/OSQocEEJnfcMkniEl8u3U8/%231483---Initial-Bitcoin-support?node-id=1182%3A63576&t=0OeOFKmcaHOATyvZ-4

**Comment by edu-stx** [comment]
@mica000 ok! merging this now, but we can def add it later.

**Pull request merged**